Islandora/islandora
# Fix/batch upload children, with validation according to default widget

**GitHub Issue**: https://github.com/Islandora/documentation/issues/2129

* Other Relevant Links (Google Groups discussion, related pull requests,
 Release pull requests, etc.)

# What does this Pull Request do?

Implements a multi-(/two-)step wizard for bulk child node (and media) creation, where:

* The first step identifies the type of media, allowing the target media field to be isolated; and,
* The second step renders the default widget for the field (with the cardinality unlocked to allow an arbitrary number of items), permitting the upload proper.

... By using the widget this way, ideally we maintain any relevant validation regarding file types and sizes and the like, but also should make use of the field's configured upload location and more.

# What's new?

Implemented the wizard(s) and deprecated the old forms.

* Does this change add any new dependencies? [`ctools` is now a dependency](https://www.drupal.org/project/ctools) 
* Does this change require any other modifications to be made to the repository
 (i.e. Regeneration activity, etc.)? No.
* Could this change impact execution of existing code? Any alterations of the old forms may need to be updated.

# How should this be tested?

Really, should be effectively regression-testing the two batch upload mechanisms being replaced with multi-step wizards:

Use the child-addition wizard to "Batch Upload Children" to create child nodes of a given node. `/node/{node}/members/upload`, should be present as a local action on the `/node/{node}/members` route... its use _should_ be relatively straight-forward?

Use the media-addition wizard to "Batch Upload Media" to create media on a given node. `/node/{node}/media/upload`, should be present as a local action on the `/node/{node}/media` route... and should be similarly straight-forward to use.

# Documentation Status

* Does this change existing behaviour that's currently documented? Possibly?
* Does this change require new pages or sections of documentation? Probably not?
* Who does this need to be documented for? 
* Associated documentation pull request(s): ___  or documentation issue ___

# Additional Notes:
Any additional information that you think would be helpful when reviewing this
 PR.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/8-x-committers
